apps/web/vibes/soul/sections/cart/client.tsx

```diff
@@ -42,6 +43,13 @@ export interface Cart<LineItem extends CartLineItem> {
   summaryItems: CartSummaryItem[];
   total: string;
   totalLabel?: string;
+  couponCodes?: string[];
+  couponCodeAction?: Action<CouponCodeFormState, FormData>;
+  couponCodeCtaLabel?: string;
+  couponCodeDisabled?: boolean;
+  couponCodeLabel?: string;
+  couponCodePlaceholder?: string;
+  couponCodeRemoveLabel?: string;
```

Comment:
I'm not sure if these props need to be part of the `cart` prop, or if they should be root props in the `Cart` section.

Reply:
They should be props in the `Cart` section, IMO. These are not properties that you would have in a cart entity in an ecommerce API, right?

Also, I'd nest these inside a `coupon` prop that's optional and use the presence of that prop to determine whether to render the `CouponCodeForm` component instead of using the presence of the action for that.

Reply:
Moving to root.